### PR TITLE
Project view command

### PR DESCRIPTION
**Pull request by cheese3660**
Implements the first part of #1311, that being a command to display information contained in the current project's `project.json` file.

It displays general information about the project, then target specific information in a list afterwards.

An example output is shown in this image.
an example output of c3c project view

The project.json for that example is as follows
```json
{
	"langrev": "1",
	"warnings": ["no-unused"],
	"linked-libraries": ["glfw"],
	"authors": [],
	"version": "0.1.0",
	"dependencies": ["opengl", "stb"],
	"dependency-search-paths": ["dependencies"],
	"sources": ["./dependencies/helpers.c3", "./dependencies/glfw.c3", "./dependencies/camera.c3"],
	"c-sources": ["./dependencies/stb.c3l/native/*"],
	"cflags": "-fPIE",
	"targets": {
		"cube": {
			"type": "executable",
			"sources": ["./examples/cube.c3"]
		},
		"triangle": {
			"type": "executable",
			"sources": ["./examples/triangle.c3"]
		},
		"texture": {
			"type": "executable",
			"sources": ["./examples/texture.c3"]
		},
		"gltf": {
			"type": "executable",
			"sources": ["./examples/gltf.c3", "./dependencies/gltf.c3"]
		}
	},
	"output": "./build",
	"symtab": 52,
	"no-entry": false
}
```

### CODE REVIEW

File: src/build/project_manipulation.c [hunk 1]
```diff
@@ -0,0 +1,311 @@
+#include "build_internal.h"
+#define PRINTFN(string, ...) fprintf(stdout, string "\n", ##__VA_ARGS__) // NOLINT
+#define PRINTF(string, ...) fprintf(stdout, string, ##__VA_ARGS__) // NOLINT
+static JSONObject* read_project() {
```
Comment (lerno) [inline src/build/project_manipulation.c:4]: This looks good bug can you just fix this to use Allman brace style so the project has a unified style?

File: src/build/project_manipulation.c [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	{
+		error_exit("Expected a map of project information in '%s'.", PROJECT_JSON);
+	}
+    return json;
+}
+
+static void print_vec(const char* header, const char** vec, bool opt) {
+	if (opt && !vec) return;
+	PRINTF("%s: ", header);
+	if (vec_size(vec) > 0) {
```
Comment (lerno) [inline src/build/project_manipulation.c:24]: Here you can use
```
FOREACH_IDX(i, char *, name, vec)
{
  if (i > 0) 
  {
    PRINTF(", ");
  }
  PRINTF("%s",vec[i]);
}
```

**Comment by cheese3660** [comment]
Alright, changed the loop and fixed brace style

**Comment by lerno** [comment]
I added some last formatting changes as well. Now we just need to check the CI

**Comment by cheese3660** [comment]
ouch the foreach loop caused some errors

**Comment by lerno** [comment]
Thank you!

**Pull request merged**